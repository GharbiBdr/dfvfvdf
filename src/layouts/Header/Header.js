import React from 'react'
import { View, TouchableOpacity } from 'react-native'
import Colors from '../../assets/styles/Colors'
import Style from '../../assets/styles/Style'
import PropTypes from 'prop-types'
import {  Icon, Label } from '../../designSystem'

const Header = (props) => {
  const renderLeftItem = () => {
    if (props.leftIconName) {
      return <Icon
        name={props.leftIconName}
        size="small"
        color={Colors.white}
        {...props.handleLeftAction && { onPress: props.handleLeftAction }}
      />
    } else {
      return <TouchableOpacity {...props.handleLeftAction && { onPress: props.handleLeftAction }}>
        <Label style={props.LeftLabelStyle}>{props.leftLabel}</Label>
      </TouchableOpacity >
    }
  }
  const renderRightItem = () => {
    if (props.rightIconName) {
      return <Icon
        name={props.rightIconName}
        size="small"
        color={Colors.white}
        {...props.handleRightAction && { onPress: props.handleRightAction }}
      />
    } else {
      return <TouchableOpacity {...props.handleRightAction && { onPress: props.handleRightAction }}>
        <Label style={props.rightLabelStyle}>{props.rightLabel}</Label>
      </TouchableOpacity >
    }
  }

  return (
    <View style={Style.headerContainer} >
      <View style={Style.hedearLeftIconContainer} >
        {renderLeftItem()}
      </View>
      <View  style={Style.hedearCenterContainer} >
        <Label style={Style.headerScreenNameLabel}>
          {props.screenName}
        </Label>
      </View >
      <View style={Style.hedearRightIconContainer}>
        {renderRightItem()}
      </View>
    </View>
  )
}

Header.defaultProps = {
  screenName: '',
  rightLabelStyle: Style.headerSecondaryLabel,
  LeftLabelStyle: Style.headerSecondaryLabel,
  rightLabel: '',
  leftLabel: '',
  leftIconName: '',
  rightIconName: ''
}

Header.propTypes = {
  screenName: PropTypes.string,
  rightLabelStyle: PropTypes.object,
  LeftLabelStyle: PropTypes.object,
  rightLabel: PropTypes.string,
  leftLabel: PropTypes.string,
  handleLeftAction: PropTypes.func,
  handleRightAction: PropTypes.func,
  leftIconName: PropTypes.string,
  rightIconName: PropTypes.string,
  navigation: PropTypes.object
}

export default Header
