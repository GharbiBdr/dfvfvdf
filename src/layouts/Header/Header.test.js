import React from 'react'
import renderer from 'react-test-renderer'
import Header from './Header'


describe('Header Test', () => {
  it('heading test snapshot', () => {
    const wrapper = renderer.create(<Header screenName="Nouveau Contact" />)
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('test icons', () => {
    const wrapper = renderer.create(<Header leftIconName="left-arrow" rightIconName="check" screenName="Home" />)
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('test press icons', () => {
    const leftPress = jest.fn()
    const rightPress = jest.fn()
    const wrapper = renderer.create(
      <Header leftIconName="left-arrow" rightIconName="check" screenName="Home" handleLeftAction={leftPress} handleRightAction={rightPress} />
    )
    wrapper.root.props.handleLeftAction()
    wrapper.root.props.handleRightAction()
    expect(leftPress.mock.calls.length).toEqual(1)
    expect(rightPress.mock.calls.length).toEqual(1)

  })

  it('test press labesl', () => {
    const leftPress = jest.fn()
    const rightPress = jest.fn()
    const wrapper = renderer.create(
      <Header leftLabel="goBack" rightLabel="check" screenName="Home" handleLeftAction={leftPress} handleRightAction={rightPress} />
    )
    wrapper.root.props.handleLeftAction()
    wrapper.root.props.handleRightAction()
    expect(leftPress.mock.calls.length).toEqual(1)
    expect(rightPress.mock.calls.length).toEqual(1)

  })
})
