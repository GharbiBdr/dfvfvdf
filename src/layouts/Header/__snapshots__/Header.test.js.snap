// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test heading test snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FF8577",
      "flexDirection": "row",
      "height": 87,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "paddingLeft": 15,
          "paddingRight": 15,
        },
        Object {
          "maxWidth": "33.333332%",
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 16,
            "fontWeight": "400",
            "height": 19,
            "letterSpacing": 1,
            "lineHeight": 18,
          }
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "paddingLeft": 15,
          "paddingRight": 15,
        },
        Object {
          "maxWidth": "33.333332%",
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Object {
          "backgroundColor": "#FF8577",
          "color": "#FFFFFF",
          "fontFamily": "Roboto-Bold",
          "fontSize": 20,
          "fontWeight": "700",
          "height": 23,
          "letterSpacing": 1,
          "lineHeight": 23,
        }
      }
    >
      Nouveau Contact
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "paddingLeft": 15,
          "paddingRight": 15,
        },
        Object {
          "maxWidth": "99.999996%",
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 16,
            "fontWeight": "400",
            "height": 19,
            "letterSpacing": 1,
            "lineHeight": 18,
          }
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`Header Test test icons 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FF8577",
      "flexDirection": "row",
      "height": 87,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "paddingLeft": 15,
          "paddingRight": 15,
        },
        Object {
          "maxWidth": "33.333332%",
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ?
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "paddingLeft": 15,
          "paddingRight": 15,
        },
        Object {
          "maxWidth": "33.333332%",
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Object {
          "backgroundColor": "#FF8577",
          "color": "#FFFFFF",
          "fontFamily": "Roboto-Bold",
          "fontSize": 20,
          "fontWeight": "700",
          "height": 23,
          "letterSpacing": 1,
          "lineHeight": 23,
        }
      }
    >
      Home
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "paddingLeft": 15,
          "paddingRight": 15,
        },
        Object {
          "maxWidth": "99.999996%",
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        î¤†
      </Text>
    </View>
  </View>
</View>
`;
